### DISCUSSION TITLE
connection problem for many-to-many prisma and mongodb

### DISCUSSION THREAD

**Discussion by harpreetk05** [prisma/prisma#18691] [Q&A] [answered]
```prisma
model owner {
  ownerId      String         @id @default(auto()) @map("_id") @db.ObjectId
   property     property[]     @relation(fields: [propertyId], references: [propertyId])
  propertyId   String[]       @db.ObjectId
  propertyrole propertyrole[]
}

model property {
  propertyId     String         @id @default(auto()) @map("_id") @db.ObjectId
   ownerId        String[]       @db.ObjectId
  owner          owner[]        @relation(fields: [ownerId], references: [ownerId])
  propertyrole   propertyrole[]
}
```
these are my relations in prisma with mongodb but while updating owner it is getting connected by property but property is not getting updated with that specific owner

**Comment by nurul3101** [comment]
Hello @harpreetk05 👋 

Can you please share the update query that you are invoking?

**Reply by harpreetk05** [reply]
CREATE API -

```ts
public async createOwner(ownerData: CreateOwnerDto): Promise<owner> {

    const mobileNumber = ownerData.mobile.toString();

    if (mobileNumber.length !== 10) {
      throw new HttpException(400, `Mobile number is not valid!`);
    }

    const mobile = Number(mobileNumber)

    const findOwner: owner = await owner.findFirst({ where: { mobile } });
    if (findOwner) throw new HttpException(409, `Your mobile number ${ownerData.mobile} already exists`);

    const createOwnerData: owner = await owner.create({ data: { ...ownerData } });
    return createOwnerData;
  }
```


UPDATE API - 
```ts
 public async updateOwner(ownerId: string, ownerData: CreateOwnerDto): Promise<owner> {
    if (isEmpty(ownerData)) throw new HttpException(400, "Invalid Owner Data");

    const findOwner: owner = await owner.findUnique({ where: { ownerId: ownerId } });
    if (!findOwner) throw new HttpException(409, "This owner doesn't exists");

    await mobileValidation(ownerData, findOwner);

    const updateOwnerData = await owner.update({ where: { ownerId: ownerId }, data: { ...ownerData } });
    return updateOwnerData;
  }
```

**Reply by nurul3101** [reply]
what is the value of ownerData that you are passing in `updateOwner` function?

**Reply by Jayakrishnan-mk** [reply]
Ownerdata is nothing but the data coming from postman. means from frontend. Just look through this:
{
    "propertyId": ["641172fbf70cf2b53c9da3b6","641176def70cf2b53c9da3ba"],
}

and the ownerId will be autogenerated.

**Reply by nurul3101** [reply]
Considering this schema:
```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

model owner {
  ownerId    String     @id @default(auto()) @map("_id") @db.ObjectId
  property   property[] @relation(fields: [propertyId], references: [propertyId])
  propertyId String[]   @db.ObjectId
  // propertyrole propertyrole[]
}

model property {
  propertyId String   @id @default(auto()) @map("_id") @db.ObjectId
  ownerId    String[] @db.ObjectId
  owner      owner[]  @relation(fields: [ownerId], references: [ownerId])
  // propertyrole propertyrole[]
}
```
Here's a script that connects owner to a property
```ts
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient({
  log: ['query', 'info', 'warn'],
});

async function main() {
  // Create a new owner
  await prisma.owner.create({
    data: {
      ownerId: '643686cf91a3270584f00220',
    },
  });

  // Create a new property
  await prisma.property.create({
    data: {
      propertyId: '643686dd981a5b7603a5a314',
    },
  });

  // update owner with property
  await prisma.owner.update({
    where: {
      ownerId: '643686cf91a3270584f00220',
    },
    data: {
      property: {
        connect: {
          propertyId: '643686dd981a5b7603a5a314',
        },
      },
    },
  });
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });
```
In this case property is updated with owner information as well

<img width="905" alt="Screenshot 2023-04-12 at 3 55 26 PM" src="https://user-images.githubusercontent.com/13049676/231430925-92b4ad23-7a5f-4a40-840e-1f7025cbb281.png">

**Comment by harpreetk05** [comment]
The same thing connection with owner and property I tried with prisma studio they are getting connected owner and property but while trying to fire the same query and data through postman or after API integration througbh frontend the owner is getting connected to property but the property is not getting connected to owner

**Reply by nurul3101** [reply]
I can't seem to reproduce this issue on my end.
I assume this is what the query looks like
```ts
const updateOwnerData = await owner.update({ where: { ownerId: ownerId }, data: { ...ownerData } });
```
And the ownerData's value is this?
```json
{
"propertyId": ["641172fbf70cf2b53c9da3b6","641176def70cf2b53c9da3ba"],
}
```
Can you try using the [set](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#set) operator to assign these propertyIds?

**Reply by nurul3101** [reply]
Do let us know in case the `set` operator works for your use case.

**Reply by nurul3101** [reply]
It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by harpreetk05** [reply]
it is not only for updating but also for creation can you tell me exactly the set operator part

**Reply by nurul3101** [reply]
@harpreetk05 Your query would look something like this:
```ts
  const updateOwnerData = await prisma.owner.update({
    where: { ownerId: mongoID },
    data: {
      propertyId: {
        set: ['641172fbf70cf2b53c9da3b6', '641176def70cf2b53c9da3ba'],
      },
    },
  });
```

**Reply by nurul3101** [reply]
Do let us know in case `set` operator works for you.

**Reply by Jayakrishnan-mk** [reply]
We tried this way. But it's connecting like same. Owner is connecting with property. But in property table that owner is not connected. So the issue is still there.

**Reply by Jayakrishnan-mk** [reply, marked as answer]
Issue solved🎈

We used this code for connecting with both tables:
const updateOwnerData = await prisma.owner.update({
    where: { ownerId: mongoID },
        data: {
           property: {
              connect: [{
                 propertyId: propertyData.propertyId
              }]
          }
     }
})

It's worked for us. 
Thank you for your valuable time. ❤️ @nurul3101 

some time this connect method is not working for other tables. that time i used this in connect query place. and it's worked:
example: owner table to project and property tables connection:

 const updateOwnerData = await owner.update({
      where: { ownerId },
      data: {
        ...ownerData,
        project: {
          connect: ownerData.projectId.map((projectId) => ({ projectId }))
        },
        property: {
          connect: ownerData.propertyId.map((propertyId) => ({ propertyId }))
        }
      }
    });

**Reply by nurul3101** [reply]
I am glad to hear that :)
Thank you so much for your kind words.
